doc/tutorials/cpu_sharing.rst

```diff
@@ -0,0 +1,142 @@
+
+  + Every thread object has an initial timeslice (ex: 10ms)
+  + timeslice is consumed with time and be counted in context switch and tick handler
+  + If timeslice is positive or zero, then switch out current thread object and put it to tail of runqueue.Then, pick next runnable one from runqueue to run.
+  + Threads who has IO request will preempt current running thread on the same pCPU.
+
+Scheduler configuration
+***********************
+
+There are two place in the code decide the usage for scheduler.
+
+* The option in Kconfig decides the only scheduler used in runtime.
+  ``hypervisor/arch/x86/Kconfig``
+
+  .. literalinclude:: ../hypervisor/arch/x86/Kconfig
+     :name: Kconfig for Scheduler
+     :caption: Kconfig for Scheduler    
+     :linenos:
+     :lines: 40-58
+     :emphasize-lines: 42
+     :language: c
+
+The default scheduler is **SCHED_NOOP**. To use the IORR, only need to set **SCHED_IORR** in **ACRN Scheduler**. 
+
+* The affinity for VMs are set in  ``hypervisor/scenarios/<scenario_name>/vm_configurations.h``
+
+   .. literalinclude:: ../hypervisor/scenarios/industry/vm_configurations.h
+     :name: Affinity for VMs
+     :caption: Affinity for VMs  
+     :linenos:
+     :lines: 31-32
+     :language: c
+
+* vCPU number corresponding to affinity is set in ``hypervisor/scenarios/<scenario_name>/vm_configurations.c`` by **vcpu_num**
+
+For example, with the below changes, pCPU0 can be shared by VM0 and VM1.
+
+      .. code-block:: none
+
+         diff --git a/hypervisor/arch/x86/Kconfig b/hypervisor/arch/x86/Kconfig
+         index 41a78fe..8ea1f9d 100644
+         --- a/hypervisor/arch/x86/Kconfig
+         +++ b/hypervisor/arch/x86/Kconfig
+         @@ -39,7 +39,7 @@ endchoice
+         
+          choice
+                 prompt "ACRN Scheduler"
+                -default SCHED_NOOP
+                +default SCHED_IORR
+                 help
+                   Select the CPU scheduler to be used by the hypervisor
+         
+         diff --git a/hypervisor/scenarios/industry/vm_configurations.h b/hypervisor/scenarios/industry/vm_configurations.h
+         index 482801a..d76510d 100644
+         --- a/hypervisor/scenarios/industry/vm_configurations.h
+         +++ b/hypervisor/scenarios/industry/vm_configurations.h
+         @@ -28,7 +28,7 @@
+                                                 "i915.enable_gvt=1 "    \
+                                                 SOS_BOOTARGS_DIFF
+         
+         -#define        VM1_CONFIG_VCPU_AFFINITY        {AFFINITY_CPU(1U)}
+         +#define        VM1_CONFIG_VCPU_AFFINITY        {AFFINITY_CPU(0U)}
+          #define        VM2_CONFIG_VCPU_AFFINITY        {AFFINITY_CPU(2U), AFFINITY_CPU(3U)}
+   
+
+From ACRN Hypervisor console:
+
+      .. code-block:: none
```

Comment:
Line 130: The code block is too far to the right. Please left-align it.

Reply:
Fixed, thanks.